File: CSharp/Tests/ExistAHigherNumberTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class ExistAHigherNumberTest
    {
        [Theory]
        [InlineData(new int[] { 5, 3, 15, 22, 4 }, 10, true)]
        [InlineData(new int[] { 1, 2, 3, 4, 5 }, 8, false)]
        [InlineData(new int[] { 4, 3, 3, 3, 2, 2, 2 }, 4, true)]
        [InlineData(new int[] { 2, 2, 2, 2 }, 5, false)]
        [InlineData(new int[] { }, 0, false)]
        public void ExistHigher_IntArrayAndIntNumInput_ReturnBooleanIfExistAtLeastOneNumberGreaterOrEqualNum(int[] arr, int n, bool expected)
        {
            var actual = ExistAHigherNumber.ExistHigher(arr, n);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/existAHigherNumber.test.js
const existHigher = require("../existAHigherNumber");

const data = [
  [[5, 3, 15, 22, 4], 10, true],
  [[1, 2, 3, 4, 5], 8, false],
  [[4, 3, 3, 3, 2, 2, 2], 4, true],
  [[2, 2, 2, 2], 5, false],
  [[], 0, false]
];

describe("existHigher", () => {
  describe("Return a boolean if there exist at least one number greater or equal to n", () => {
    test.each(data)("existHigher(%p, %i) = %s", (arr, n, result) => {
      let actual = existHigher(arr, n);

      expect(actual).toEqual(result);
    });
  });
});
